src/canonmn_int.cpp

```diff
@@ -2260,7 +2266,15 @@ constexpr TagDetails canonRFLensType[] = {{0, N_("n/a")},
                                           {276, N_("Canon RF 100-500mm F4.5-7.1L IS USM + RF1.4x")},
                                           {277, N_("Canon RF 100-500mm F4.5-7.1L IS USM + RF2x")},
                                           {278, N_("Canon RF 70-200mm F4L IS USM")},
-                                          {280, N_("Canon RF 50mm F1.8 STM")}};
+                                          {280, N_("Canon RF 50mm F1.8 STM")},
```

Comment:
@kmilos:
It is only a suggestion but perhaps you could remove the `N_( )` translation functions from all the `canonRFLensType` entries. As far as I understand it, these strings do not need translating.

Removing the functions is consistent with the `canonCsLensType` array and the other makernotes' lens values.

Reply:
Thanks - although I didn't introduce those originally, I have nothing against removing them. Mind you, the gettext N_() macro is a noop and is not incorrect here (i.e. actually explicitly prevents any extraction for translation).

Reply:
@kmilos:
> Thanks - although I didn't introduce those originally, I have nothing against removing them. Mind you, the gettext N_() macro is a noop and is not incorrect here (i.e. actually explicitly prevents any extraction for translation).


I think that using `N_( )` declares which text *is* made available for translation (assuming that you are compiling with `EXIV2_ENABLE_NLS` set to `ON`). The `N_( )` definition can be found in [src/i18n.h](https://github.com/Exiv2/exiv2/blob/7ebf2a184e28bf638a39aa30ed3784c5035b668a/src/i18n.h).

For example, the results can be see with the description string:

https://github.com/Exiv2/exiv2/blob/7ebf2a184e28bf638a39aa30ed3784c5035b668a/src/canonmn_int.cpp#L1967-L1969

which is translated in:

https://github.com/Exiv2/exiv2/blob/7ebf2a184e28bf638a39aa30ed3784c5035b668a/po/de.po#L627-L633

If `N_( )` is removed from the RF lenses then this prevents them from being passed to the translators. Including this is only a suggestion though and it is completely fine to leave it out.

As an aside, we probably have other existing text where the same idea could be applied but reviewing every string is a **HUGE** job! I have spotted a couple of small changes but have no plans to review everything.

Reply:
Btw, those examples you gave do not come from the same source file: tranlatable is extracted from `src/canonmn.cpp:1230`, while the string you showed is from `canonmn_int.cpp`.

Reply:
@kmilos:

> N_() is no-op, it is definitely not extracted for translation, unlike _(), which is.

Looking again at this, I can see you point. I was misled by the `.po` files containing the correct translations strings. Although I don't use it myself, as far as I know, the translation system does work.

I can see that we do use some `_( )` but most references seem to be `N_( )`. In the past I have always used whatever function was already in the code.

Reply:
@kmilos:
No problem, thanks for checking.

Reply:
Thanks for making me do it ;)